### PR TITLE
react-checkbox: Updating Spec and adding upgrade guide

### PR DESCRIPTION
**Pull request by sopranopillow**
This PR adds the following changes:
* Updates spec to match implementation.
* Adds README information.
* Adds `v0->v9` and `v8->v9` upgrade guides.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `179.701 kB`<br />`50.147 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `34.066 kB`<br />`11.089 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/d1112b254407f0f0c27119326ec86d2112bc6d70'>d1112b254407f0f0c27119326ec86d2112bc6d70</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22952/perf-test-react-components/Avatar-mount_base.html">1081</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22952/perf-test-react-components/Avatar-mount.html">1081</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22952/perf-test-react-components/Button-mount_base.html">632</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22952/perf-test-react-components/Button-mount.html">677</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22952/perf-test-react-components/FluentProvider-mount_base.html">2218</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22952/perf-test-react-components/FluentProvider-mount.html">2285</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22952/perf-test-react-components/FluentProviderWithTheme-mount_base.html">330</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22952/perf-test-react-components/FluentProviderWithTheme-mount.html">345</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22952/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">289</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22952/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">286</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22952/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">362</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22952/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">357</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22952/perf-test-react-components/MakeStyles-mount_base.html">1903</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22952/perf-test-react-components/MakeStyles-mount.html">1898</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by GeoffCoxMSFT — APPROVED** [review]

File: apps/public-docsite-v9/src/Concepts/Upgrade/FromV0/Components/Checkbox.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+The main difference between v0 and v9 `Checkbox` is that v9 offers a circular variant and does not provide a toggle prop.
+
+## Examples
+
+### Basic Upgrade
+
+Basic usage of `Checkbox` v0
+
+```tsx
+import React from 'react';
```
Comment (spmonahan) [inline apps/public-docsite-v9/src/Concepts/Upgrade/FromV0/Components/Checkbox.stories.mdx:18]: nit: should be `import * as React from 'react';`

File: apps/public-docsite-v9/src/Concepts/Upgrade/FromV0/Components/Checkbox.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+| `className`      | `className`      |                                                             |
+| `defaultChecked` | `defaultChecked` | Mutually exclusive with `checked`                           |
+| `design`         | n/a              |                                                             |
+| `disabled`       | `disabled`       |                                                             |
+| `indicator`      | `indicator`      |                                                             |
+| `key`            | `key`            |                                                             |
+| `label`          | `label`          |                                                             |
+| `labelPosition`  | `labelPosition`  |                                                             |
+| `onChange`       | `onChange`       |                                                             |
+| `onClick`        | `key`            | v9 uses the intrinsic React prop                            |
```
Comment (spmonahan) [inline apps/public-docsite-v9/src/Concepts/Upgrade/FromV0/Components/Checkbox.stories.mdx:67]: I think the v9 column should be `onClick` and the "notes" column should be for the `key` row above.

**Review by spmonahan — CHANGES_REQUESTED** [review]

**Review by khmakoto — APPROVED** [review]

**Review by spmonahan — APPROVED** [review]

**Pull request merged**